src/main/java/QueryServlet.java

```diff
@@ -218,12 +244,128 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
                     "This query is not supported by census data. Try asking a more general one."));
         return;
       }
+
+      String formattedData;
+      try {
+        formattedData = reformatDataArray(data, dataRow, !location.equals("state"));
+      } catch (InvalidObjectException e) {
+        response.setStatus(HttpServletResponse.SC_BAD_GATEWAY);
+        response.setContentType("application/json;");
+        response
+            .getWriter()
+            .println(sendError("An error occurred while trying to retrieve census data."));
+        return;
+      }
+
       JsonObject jsonResponse = new JsonObject();
-      Gson gson = new Gson();
-      jsonResponse.addProperty("censusData", data);
+      jsonResponse.addProperty("censusData", formattedData);
       jsonResponse.addProperty("tableLink", censusTableLink);
       response.setContentType("application/json;");
       response.getWriter().println(jsonResponse.toString());
     }
   }
+
+  // Some data arrays need reformatting (columns added together, etc.) before they can be
+  // visualized.
+
+  // Math.round returns a long, but we can cast to int (max value ~2 billion) because our numbers
+  // will always be smaller than the U.S. population (~330 million)
+  BiFunction<Float, Float, Integer> add = (Float a, Float b) -> (int) Math.round(a + b);
+  BiFunction<Float, Float, Integer> rightSubtract = (Float a, Float b) -> (int) Math.round(a - b);
+  BiFunction<Float, Float, Integer> percent =
+      (Float a, Float b) -> (int) Math.round((a / 100.0) * b);
+
+  Map<String, List<BiFunction>> dataRowToReformatFunction =
+      ImmutableMap.<String, List<BiFunction>>builder()
+          .put("DP05_0001E,DP05_0018E", ImmutableList.of(rightSubtract))
```

Comment:
Never worked with BiFunctions so I'm not entirely sure how these lines work. Are you mapping the two data tables to the appropriate function that will eventually be used to calculate the population value? Eg: On this line you map "DP05_0001E, DP05_0018E" to the function rightSubtract because we eventually want to subtract DP05_0018E from DP050001E?

Reply:
Yes, exactly! A Function (of which a BiFunction is a specific type) is just a way of passing a normal function/method as an object. Later in the code, I can call rightSubtract.apply(someNumber, someOtherNumber), and it will apply the function I wrote, (a, b) -> a - b, to subtract someOtherNumber from someNumber.
In this case, 0001E is the total population and 0018E is the adult population. So the subtraction will give us the child population.